Add method to check whether WorkspaceDefaultMembers has a value

foresterre opened the pull request:
This method checks whether the called `cargo-metadata` supports reading the workspace default members, which is false for versions of Cargo prior to 1.71

Alternatively, I could:
- add an Option variant method like `workspace_default_packages` directly on `Metadata`: `pub fn workspace_default_packages(&self) -> Option<Vec<&Package>>`, or
- reverse the name from `is_available` to `is_missing` (which would also allow the removal of the `doc(hidden)` function `workspace_default_members_is_missing`).


fixes #254

oli-obk commented:
> * reverse the name from `is_available` to `is_missing` (which would also allow the removal of the `doc(hidden)` function `workspace_default_members_is_missing`).

is this test usually going to be used as `if !foo.is_available() {...}` or as `if foo.is_available() {}`? Oh to be honest we could just add both, just like `Option` has `is_none` and `is_some`.

foresterre commented:
I've updated the PR to include both. 

I've also removed the `doc(hidden) workspace_default_members_is_missing`, which is no longer necessary when the is_missing variant exists. It was both marked hidden and noted that it "might be removed in the future". Are you okay with its removal?
